pkg/parsers/operatingsystem/operatingsystem_windows.go

```diff
@@ -1,44 +1,40 @@
 package operatingsystem // import "github.com/docker/docker/pkg/parsers/operatingsystem"
 
 import (
-	"unsafe"
+	"fmt"
 
-	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
 )
 
-// See https://code.google.com/p/go/source/browse/src/pkg/mime/type_windows.go?r=d14520ac25bf6940785aabb71f5be453a286f58c
-// for a similar sample
-
 // GetOperatingSystem gets the name of the current operating system.
 func GetOperatingSystem() (string, error) {
 
-	var h windows.Handle
-
 	// Default return value
 	ret := "Unknown Operating System"
 
-	if err := windows.RegOpenKeyEx(windows.HKEY_LOCAL_MACHINE,
-		windows.StringToUTF16Ptr(`SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\`),
-		0,
-		windows.KEY_READ,
-		&h); err != nil {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\WIndows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		return ret, err
+	}
+	defer k.Close()
+
+	pn, _, err := k.GetStringValue("ProductName")
+	if err != nil {
 		return ret, err
 	}
-	defer windows.RegCloseKey(h)
-
-	var buf [1 << 10]uint16
-	var typ uint32
-	n := uint32(len(buf) * 2) // api expects array of bytes, not uint16
-
-	if err := windows.RegQueryValueEx(h,
-		windows.StringToUTF16Ptr("ProductName"),
-		nil,
-		&typ,
-		(*byte)(unsafe.Pointer(&buf[0])),
-		&n); err != nil {
+	ret = pn
+
+	ri, _, err := k.GetStringValue("ReleaseId")
+	if err != nil {
+		return ret, err
+	}
+	ret = fmt.Sprintf("%s Version %s", ret, ri)
+
+	ubr, _, err := k.GetIntegerValue("UBR")
+	if err != nil {
 		return ret, err
 	}
-	ret = windows.UTF16ToString(buf[:])
+	ret = fmt.Sprintf("%s, Update Build Revision %d", ret, ubr)
```

Comment:
The format looks okay, and should look like this for Windows Server 1709:
`Windows Server Standard Version 1709, Update Build Revision 248`

However, did you consider something more consistent with `winver.exe`?
`Windows Server Standard Version 1709 (OS Build 16299.248)`

Reply:
Sure updated.

Client (RS4 Pre-release)

```
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 17114 (17114.1000.amd64fre.rs4_release_base.180301-1503)
Operating System: Windows 10 Enterprise Insider Preview Version 1803 (OS Build 17114.1000)
OSType: windows
Architecture: x86_64
CPUs: 8
Total Memory: 15.93GiB
```

Reply:
And server RS3

```
Default Isolation: process
Kernel Version: 10.0 16299 (16299.248.amd64fre.rs3_release_svc_escrow.180209-1727)
Operating System: Windows Server Datacenter Version 1709 (OS Build 16299.248)
OSType: windows
Architecture: x86_64
CPUs: 32
Total Memory: 16.01GiB
```